### PR TITLE
solves issue #22929 by adding knowledge checks and links to Ruby/Test Driven Development lesson

### PR DESCRIPTION
**Pull request by GitHubBlues**
- [ x] You have read [The Odin Projects Contributing Guide](https://github.com/TheOdinProject/curriculum/blob/main/CONTRIBUTING.md).
 - [x ] The title summarizes the change and where it happened, for example: "Fixes punctuation in Clean Code lesson".
 - [ ] If the PR is related to an open issue, use a [relevant keyword](https://docs.github.com/en/github/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests) and reference it with the `#` sign and the issue number.
- [x ] If changes are requested, please make the changes in a timely manner. After you submit the changes, request another review from the maintainer (top right).

#### 1.Describe the changes made and include why they are necessary or important:

Added three knowledge checks with the corresponding links as was requested for solving issue#22929

#### 2. Related Issue

keyword #XXXXX

### CODE REVIEW

File: ruby_programming/testing_with_rspec/test_driven_development.md [hunk 1]
```diff
@@ -40,3 +40,12 @@ As usual, it depends. Still, here are some reasons we think it might be importan
 2. For a more rounded take, consider DHH's ["TDD is Dead. Long Live Testing"](http://david.heinemeierhansson.com/2014/tdd-is-dead-long-live-testing.html), which sparked controversial conversations on what TDD is, what it has become, and what it should be.
 
 </div>
+
+### Knowledge Check
+
+This section contains questions for you to check your understanding of this lesson. If you’re having trouble answering the questions below on your own, review the material above to find the answer.
+
+- <a class="knowledge-check-link" href="https://www.theodinproject.com/paths/full-stack-ruby-on-rails/courses/ruby-programming/lessons/test-driven-development#what-is-test-driven-development">What is the "red-green-refactor" cycle?</a>
```
Comment (wise-king-sullyman) [inline ruby_programming/testing_with_rspec/test_driven_development.md:48]:
```suggestion
- <a class="knowledge-check-link" href="#what-is-test-driven-development">What is the "red-green-refactor" cycle?</a>
```

File: ruby_programming/testing_with_rspec/test_driven_development.md [hunk 2]
```diff
@@ -40,3 +40,12 @@ As usual, it depends. Still, here are some reasons we think it might be importan
...
+- <a class="knowledge-check-link" href="https://thoughtbot.com/blog/rails-test-types-and-the-testing-pyramid#unit-tests">What is the difference between unit and integration tests?</a>
```
Comment (wise-king-sullyman) [inline ruby_programming/testing_with_rspec/test_driven_development.md:49]: I see that this is where the link on line 25 now goes, can you update it there?

**Review by wise-king-sullyman — CHANGES_REQUESTED** [review]
This looks really good! I just have a couple of small changes I'd like to see.

File: ruby_programming/testing_with_rspec/test_driven_development.md (hunk 2 above)
Comment (GitHubBlues) [reply ruby_programming/testing_with_rspec/test_driven_development.md:49]: Yes, I will do that.

Comment (GitHubBlues) [reply ruby_programming/testing_with_rspec/test_driven_development.md:49]: How to I update the link? Here on GitHub or on my local repository? Do I need a new branch?

Comment (GitHubBlues) [reply ruby_programming/testing_with_rspec/test_driven_development.md:49]: Do I have to make a new PR?

Comment (wise-king-sullyman) [reply ruby_programming/testing_with_rspec/test_driven_development.md:49]: No don't make a new PR, you can make changes on your local copy of the repo and push them to the same branch that you're doing the PR from, or you can make the edits from GitHub directly, again just make sure the changes go to your `feature/knowledgeChecks` branch of your fork.

Comment (GitHubBlues) [reply ruby_programming/testing_with_rspec/test_driven_development.md:49]: I am a bit confused now. So how should the link be:

href="https://www.theodinproject.com/paths/full-stack-ruby-on-rails/courses/ruby-programming/lessons/test-driven-development#what-is-test-driven-development"

or

href="#what-is-test-driven-development"
?

Comment (GitHubBlues) [reply ruby_programming/testing_with_rspec/test_driven_development.md:49]: I decided that it had to be the second option. 
I fixed it and hope it is OK. 
Thanks for your help. I learned a lot!

Comment (wise-king-sullyman) [reply ruby_programming/testing_with_rspec/test_driven_development.md:49]: Looks like you addressed the first change I requested, but the link on line 25 is still not the same URL used in your Knowledge Check on line 49, can you update the line 25 link to match your line 49 link?

If I'm being unclear here please let me know.

File: ruby_programming/testing_with_rspec/test_driven_development.md (hunk 1 above)
Comment (GitHubBlues) [reply ruby_programming/testing_with_rspec/test_driven_development.md]: I had not realized there were two suggested changes ..
I fixed it!

**Comment by wise-king-sullyman** [comment]
@GitHubBlues I think we're speaking past each other a bit.

image
  
This link here on line 25 redirects to https://thoughtbot.com/blog/rails-test-types-and-the-testing-pyramid, which is the link you correctly used in your Knowledge Check.

I'd like to see this redirect link be the link used in the place that I put the box around on line 25, so that the links will be identical in both places in the lesson.

**Comment by GitHubBlues** [comment]
I am sorry this has taken so long. 
Hope its OK now.

**Comment by CouchofTomato** [comment]
@wise-king-sullyman Can you please review when you have time

**Review by wise-king-sullyman — APPROVED** [review]
Looks perfect! Thank you for your contribution!

**Pull request merged**